client/client.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/fabric8-services/fabric8-auth-client/auth"
+	goaclient "github.com/goadesign/goa/client"
+	"github.com/goadesign/goa/middleware/security/jwt"
+	errs "github.com/pkg/errors"
+)
+
+func NewAuthClient(hostURL string) (*AuthClient, error) {
+	u, err := url.Parse(hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	client := http.Client{}
+	c := auth.New(&doer{
+		target: goaclient.HTTPClientDoer(&client),
+	})
+	c.Host = u.Host
+	c.Scheme = u.Scheme
+	return &AuthClient{c}, nil
+}
+
+type doer struct {
+	target goaclient.Doer
+}
+
+func (d *doer) Do(ctx context.Context, req *http.Request) (*http.Response, error) {
+	token := jwt.ContextJWT(ctx)
+	if token != nil {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Raw))
+	}
+	return d.target.Do(ctx, req)
+}
+
+type AuthClient struct {
+	*auth.Client
+}
+
+func (c *AuthClient) CheckSpaceScope(ctx context.Context, spaceID, requiredScope string) error {
```

Comment:
there are few variety that is possible, like whether one want to use user_token/sa_token, whether one want to have httpcache or not, whether one want to check one `requiredScope` or more.  Also, as we add more methods to this AuthClient, those new methods will have some more variety.  So for now, not moving to common is also fine from MPOV.

Reply:
It must be in some shared lib (like common). Really. Otherwise we will start duplicating this code across multiple repos very soon.
I don’t know if we ever will need to check scopes by service accounts or for multiple scopes.
But the case where we need to check a scope of some resource for the user token is fundamental. It was agreed on the very beginning when we just started to implement AuthZ that we would need to provide some shared lib for that so individual services wouldn’t need to reimplement it everywhere.

Reply:
done https://github.com/fabric8-services/fabric8-common/pull/62